Sources/App/Commands/Analyze.swift

```diff
@@ -371,23 +406,44 @@ func diffVersions(client: Client,
         return Current.reportError(client, .error, appError)
             .transform(to: [])
     }
-    
+
     let references = tags.map { tags in [defaultBranch].compactMap { $0 } + tags }
-    let incoming: EventLoopFuture<[Version]> = references
+    return references
         .flatMapEachThrowing { ref in
             let revInfo = try Current.git.revisionInfo(ref, cacheDir)
             let url = package.versionUrl(for: ref)
             return try Version(package: package,
                                commit: revInfo.commit,
                                commitDate: revInfo.date,
                                reference: ref,
-                               url: url) }
-    
-    return Version.query(on: transaction)
-        .filter(\.$package.$id == pkgId)
-        .all()
-        .and(incoming)
-        .map(Version.diff)
+                               url: url)
+        }
+}
+
+
+func throttle(lastestExistingVersion: Version?, incoming: [Version]) -> [Version] {
+    guard let existingVersion = lastestExistingVersion,
+          let latestExisting = existingVersion.commitDate else {
+        // there's no existing branch version -> leave incoming alone (which will lead to addition)
+        return incoming
+    }
+
+    guard let incomingVersion = incoming.latestBranchVersion,
+          let latestIncoming = incomingVersion.commitDate else {
+        // there's no incoming branch version -> leave incoming alone (which will lead to removal)
+        return incoming
+    }
+
+    #warning("distance to latestIncoming to to Current.date() ?")
+    let delta = latestIncoming.timeIntervalSinceReferenceDate - latestExisting.timeIntervalSinceReferenceDate
```

Comment:
In practise this should make much of a difference as we're discovering new versions every couple of hours at most. I.e. there won't really be a meaningful difference between when a new version appears and `Current.date()`

I think we should go with `Current.date()`.

Reply:
As discussed, change to `Current.date()`

Reply:
Fixed